tests/sentry/deletions/test_project.py

```diff
@@ -212,3 +223,44 @@ def test_delete_with_uptime_monitors(self):
         assert not ProjectUptimeSubscription.objects.filter(
             id=project_uptime_subscription.id
         ).exists()
+
+    def test_delete_with_workflow_engine_models(self):
+        project = self.create_project(name="test")
+
+        data_condition_group = self.create_data_condition_group()
+        data_condition = self.create_data_condition(condition_group=data_condition_group)
+        snuba_query = self.create_snuba_query()
+        subscription = QuerySubscription.objects.create(
+            project=project,
+            status=QuerySubscription.Status.ACTIVE.value,
+            subscription_id="123",
+            snuba_query=snuba_query,
+        )
+        data_source = self.create_data_source(
+            organization=self.organization, source_id=subscription.id
+        )
+        detector = self.create_detector(
+            project_id=project.id,
+            name="Test Detector",
+            type=MetricIssue.slug,
+            workflow_condition_group=data_condition_group,
+        )
+        workflow = self.create_workflow()
+        data_source_detector = self.create_data_source_detector(
+            data_source=data_source, detector=detector
+        )
+        detector_workflow = DetectorWorkflow.objects.create(detector=detector, workflow=workflow)
```

Comment:
fyi - this can be covered with https://github.com/getsentry/sentry/blob/master/tests/sentry/workflow_engine/test_base.py#L118-L157

```py
workflow, detector, detector_workflow, workflow_triggers = self.create_detector_and_workflow()
```
This will also create those data condition groups for the workflow_condition_group and the when_condition_group

Reply:
I updated to use this but it seems like this helper wasn't made for metric issues even though the default `detector_type` is `MetricIssue.slug`. I had to make some changes and create the data condition group and data condition separately. Maybe a separate refactor would help here.

Reply:
👍 separating it sounds good, i'm not feeling strongly about needing to make this change, just wanted to give ya a heads up that we had some helper methods.